common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java

```diff
@@ -6,53 +6,41 @@
 
 import com.wynntils.core.WynntilsMod;
 import com.wynntils.core.managers.CoreManager;
-import com.wynntils.mc.utils.ComponentUtils;
 import com.wynntils.mc.utils.McUtils;
 import com.wynntils.wynn.event.QuestBookReloadedEvent;
-import com.wynntils.wynn.event.WorldStateEvent;
-import com.wynntils.wynn.model.WorldStateManager;
 import com.wynntils.wynn.model.container.ContainerContent;
 import com.wynntils.wynn.model.container.ScriptedContainerQuery;
 import com.wynntils.wynn.utils.ContainerUtils;
 import com.wynntils.wynn.utils.InventoryUtils;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.util.Objects;
 import net.minecraft.ChatFormatting;
 import net.minecraft.network.chat.TextComponent;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
-import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class QuestBookManager extends CoreManager {
-    private static final Pattern QUEST_NAME_MATCHER = Pattern.compile("^§.§l([^֎À]*)[֎À]+ (§e\\[Tracked\\])?$");
     private static final int NEXT_PAGE_SLOT = 8;
 
     private static List<QuestInfo> quests = List.of();
     private static List<QuestInfo> newQuests;
 
     public static void init() {}
 
-    @SubscribeEvent
-    public static void onQuestsReloaded(QuestBookReloadedEvent event) {
-        McUtils.player().sendMessage(new TextComponent("Quest Book scanned.").withStyle(ChatFormatting.GRAY), null);
-    }
-
-    @SubscribeEvent
-    public static void onWorldChange(WorldStateEvent e) {
-        if (e.getNewState() == WorldStateManager.State.WORLD) {
-            rescanQuestBook();
-        }
-    }
-
     /**
      * Trigger a rescan of the quest book. When the rescan is done, a QuestBookReloadedEvent will
      * be sent. The available quests are then available using getQuests.
      */
     private static void queryQuestBook() {
         ScriptedContainerQuery.QueryBuilder queryBuilder = ScriptedContainerQuery.builder("Quest Book Query")
-                .onError(msg -> WynntilsMod.warn("Problem querying Quest Book: " + msg))
+                .onError(msg -> {
```

Comment:
There is also an .onError in trackQuest that could use the same code of doing both log and chat. (Perhaps we should have a convenience method for that in QueryBuilder?)

Reply:
What's the difference?

Reply:
Sorry if I was not really clear. We have two instances of the queryBuilder in this class. I think both of them might benefit from having an error handler that prints error message both to the log and to the chat. (Basically copying this code to `trackQuest` as well)

Then I thought copying code is bad, and having an error handler sending the same message both to the log and the chat might be a common scenario for when a scripted container query fails, so maybe we should have a special support method for that, like `.onErrorReport(String msg)` that does this. But that was more of an idea for future enhancements on the container query api.

Reply:
I see. Yes, this is a nice idea. (but if we print similar messages for different queries, we cannot easily tell what happened)

Reply:
So, my idea is that we take in a specific "prefix" message when calling it, and then append the error message. Like here `"Problem querying Quest Book: " + msg`, so we'd call this as `.onErrorReport("Problem querying Quest Book")`